packages/components/carousel/src/Carousel.tsx

```diff
@@ -1,89 +1,162 @@
     }
   };
 
+  const prevSlide = () => {
+    if (!isTransitioning) {
+      setIsTransitioning(true);
+      setCurrentSlide((prevSlide: number) => prevSlide - 1);
+    }
+  };
+
+  useEffect(() => {
+    if (autoplay) {
+      const interval = setInterval(nextSlide, autoplayInterval);
+      return () => clearInterval(interval);
+    }
+  }, [autoplay, autoplayInterval]);
+
   return (
-    <CarouselStyled>
-      <ResponsiveCarousel
-        showArrows={showArrows}
-        showStatus={showStatus}
-        showIndicators={showIndicators}
-        showThumbs={showThumbs}
-        infiniteLoop={infiniteLoop}
-        autoPlay={autoPlay}
-        interval={interval}
-        centerMode={true}
-        centerSlidePercentage={35}
-        {...props}
-        aria-label="Image and video carousel"
-        renderIndicator={(
-          clickHandler: MouseEventHandler<HTMLButtonElement>,
-          isSelected: boolean,
-          index: number,
-        ) => (
-          <IndicatorItem key={index}>
-            <IndicatorButton
-              type="button"
-              aria-pressed={isSelected}
-              aria-label={`Slide ${index + 1}`}
-              onClick={clickHandler}
-              isSelected={isSelected}
-            >
-              {isSelected ? '●' : '○'}
-            </IndicatorButton>
-          </IndicatorItem>
-        )}
+    <CarouselStyled
+      styled={mergedThemeValues as Required<CarouselStyledProps>}
+      {...props}
+    >
+      <DraggableArea
+        data-testid="draggable-area"
+        onMouseDown={(event: React.MouseEvent<HTMLDivElement, MouseEvent>) =>
+          utilsHandleDragStart(event, dragStartX)
+        }
+        onMouseUp={(event: React.MouseEvent<HTMLDivElement, MouseEvent>) =>
+          utilsHandleDragEnd(event, dragStartX, nextSlide, prevSlide)
+        }
+      />
+      <CarouselTrack
+        styled={mergedThemeValues as Required<CarouselStyledProps>}
+        totalSlides={infiniteLoop ? totalSlides + 2 : totalSlides}
+        currentSlide={currentSlide}
+        visibleSlides={visibleSlides}
+        onTransitionEnd={handleTransitionEnd}
+        ref={trackRef}
       >
-        {contents.map((content: Content, index: number) => (
-          <Slide key={`carousel-container-${index}`}>
-            {renderContent(content)}
-          </Slide>
+        {infiniteLoop && (
+          <CarouselSlide key={`clone-last-${slides[totalSlides - 1].children}`}>
+            {slides[totalSlides - 1].children}
+          </CarouselSlide>
+        )}
+        {slides.map((slide: CarouselSlideProps, index: number) => (
+          <CarouselSlide key={`slide-${index}`}>{slide.children}</CarouselSlide>
         ))}
-      </ResponsiveCarousel>
+        {infiniteLoop && (
+          <CarouselSlide key={`clone-first-${slides[0].children}`}>
+            {slides[0].children}
+          </CarouselSlide>
+        )}
+      </CarouselTrack>
+
+      {showArrows && (
+        <>
+          <ArrowButton
```

Comment:
The arrows are too small

Reply:
needs to be changed when there is design with the right icons